packages/tsx-editor/scripts/copyTypes.js

```diff
@@ -0,0 +1,47 @@
+// @ts-check
+
+const { just } = require('@uifabric/build');
+const { copyTask } = just;
+const { expandSourcePath } = require('@uifabric/build/tasks/copy');
+const fs = require('fs');
+const path = require('path');
+
+// Copy office-ui-fabric-react.d.ts and other .d.ts files it references to the dist folder,
+// so we can more easily load them into the editor later.
+module.exports = function copyTypes() {
+  const packagesToResolve = ['office-ui-fabric-react', '@uifabric/example-data'];
+  const resolvedPackages = [];
+  const pathsToCopy = [];
+
+  while (packagesToResolve.length) {
+    const package = packagesToResolve.shift();
+    resolvedPackages.push(package);
+
+    const packageMatch = package.match(/^(@uifabric\/)?([\w-]+)/);
+    const dtsPath = expandSourcePath(`${package}/dist/${packageMatch[2]}.d.ts`);
+
+    if (fs.existsSync(dtsPath)) {
+      // copy this .d.ts
+      pathsToCopy.push(dtsPath);
+
+      // add any other @uifabric packages it references for processing (ignore React imports and other imports)
+      const dtsContents = fs.readFileSync(dtsPath).toString();
+      const importRegex = /import .*? from '(@uifabric\/[\w-]+)/gm;
```

Comment:
Not maybe for this change, but since we're already using typescript... we could use typescript to parse for this with a simple visitor?

An example of parsing and doing something with the AST here: https://github.com/kenotron/riceburn/blob/master/src/typescript/index.ts

Reply:
Oh interesting (and wish I'd thought of it sooner)... Trouble is I'm not sure whether any of the APIs provide access to either a raw TS object or the syntax tree.

Reply:
Looked at the monaco-typescript code and the worker internally creates a TS language service instance but doesn't make it available to consumers.
https://github.com/microsoft/monaco-typescript/blob/master/src/tsWorker.ts